build_modules/lib/src/meta_module.dart

```diff
@@ -252,14 +252,15 @@ MetaModule _coarseModulesForLibraries(
 
 MetaModule _fineModulesForLibraries(
     AssetReader reader, List<ModuleLibrary> libraries, DartPlatform platform) {
-  var modules = libraries
-      .map((library) => Module(
+  var modules = [
+    for (var library in libraries)
+      Module(
           library.id,
-          library.parts.followedBy([library.id]),
+          [...library.parts, library.id],
```

Comment:
This is one place where I'm not fully confident that the eager collection won't be less performant than a lazy Iterable.

The `Module` constructor will do `UnmodifiableSetView(HashSet.of(source))` so it's going to be iterated and copied into a new collection immediately. Since these are small and we're allocating a list for `[library.id]` anyway I don't expect this to be impactful, but I might be overlooking something.

Reply:
Ya it is something you would have to measure to really know. I agree it is very unlikely to matter though, given these are small lists, and we aren't creating any extra list allocations. The majority of files have no parts anyways so the list will be identical. LGTM